Source/SwiftLintFramework/Extensions/Array+SwiftLint.swift

```diff
@@ -42,4 +43,44 @@ extension Array {
             return dictionary
         }
     }
+
+    public func parallelForEach(block: @escaping ((Element) -> Void)) {
+        _ = parallelMap(transform: block)
+    }
+
+    func parallelFlatMap<T>(transform: @escaping ((Element) -> [T])) -> [T] {
+        return parallelMap(transform: transform).flatMap { $0 }
+    }
+
+    func parallelMap<T>(transform: @escaping ((Element) -> T)) -> [T] {
+        let count = self.count
+        let maxConcurrentJobs = ProcessInfo.processInfo.activeProcessorCount
+
+        guard count > 1 && maxConcurrentJobs > 1 else {
+            // skip GCD overhead if we'd only run one at a time anyway
+            return map(transform)
+        }
+
+        var result = [(Int, [T])]()
+        result.reserveCapacity(count)
+        let group = DispatchGroup()
+        let uuid = NSUUID().uuidString
+        let jobCount = Int(ceil(Double(count) / Double(maxConcurrentJobs)))
+
+        let queueLabelPrefix = "io.realm.SwiftLintFramework.map.\(uuid)"
```

Comment:
I didn't use it because I wasn't aware of it! Thanks for sharing 😊 

The docs are non-existent, unfortunately: https://developer.apple.com/reference/dispatch/dispatchqueue/2016088-concurrentperform

But doing a bit of digging, the Dispatch Swift overlay [seems to implement that function](https://github.com/apple/swift-corelibs-libdispatch/blob/012f48b05b0aa62b95f034236e98685175fc4d2c/src/swift/DispatchStubs.cc#L75) with `dispatch_apply`, which _is_ well documented here: https://developer.apple.com/reference/dispatch/1452846-dispatch_apply_f

It seems to be exactly what I've reimplemented here, so I think I'll refactor to use that instead.